File: src/week1/P003.py
# 11659
# 합 배열 만드는 공식
# A[i] 부터 A[j] 까지의 배열 합
# S[i] = S[i-1] + A[i]
# 구간 합 구하는 공식
# S[j] - S[i-1]
import sys
input = sys.stdin.readline
n, m = map(int, input().split())
numList = list(map(int, input().split()))
prev = [0]
result = 0

for i in range(n):
    result += numList[i]
    prev.append(result)

for num in range(m):
    i, j = map(int, input().split())
    print(prev[j]-prev[i-1])

File: src/week1/P003.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m] = input[0].split(' ').map(Number);
const numbers = input[1].split(' ').map(Number);

// 구간 합 배열 생성
const prefixSum = new Array(n + 1).fill(0);
for (let i = 0; i < n; i++) {
    prefixSum[i + 1] = prefixSum[i] + numbers[i];
}

const results = [];
for (let k = 2; k < m + 2; k++) {
    const [i, j] = input[k].split(' ').map(Number);
    // 구간 합 공식: S[j] - S[i-1]
    const result = prefixSum[j] - prefixSum[i - 1];
    results.push(result);
}

console.log(results.join('\n'));